collector/collector.go

```diff
@@ -54,43 +53,31 @@ func getConfig() string {
 }
 
 func NewCollector(factories component.Factories) *Collector {
-	f, err := os.Open(configFile)
-	if err != nil {
-		log.Printf("Reading AOT config from file: %v failed.\n", configFile)
-		panic("Cannot load Collector config.")
-	}
-	var r io.Reader = f
 	col := &Collector{
 		factories:      factories,
-		parserProvider: parserprovider.NewInMemory(r),
+		mapProvider: parserprovider.NewExpandMapProvider(parserprovider.NewFileMapProvider(getConfig())),
 	}
 	return col
 }
 
-func (c *Collector) Start() error {
+func (c *Collector) Start(ctx context.Context) error {
 	params := service.CollectorSettings{
 		BuildInfo: component.BuildInfo{
 			Command:  "otelcol",
 			Description: "Lambda Collector",
 			Version:  Version,
 		},
-		ParserProvider: c.parserProvider,
+		ConfigMapProvider: c.mapProvider,
 		Factories:      c.factories,
 	}
 	var err error
 	c.svc, err = service.New(params)
 	if err != nil {
 		return err
 	}
-	cmd := service.NewCommand(c.svc)
-	if args, ok := os.LookupEnv("OPENTELEMETRY_COLLECTOR_ARGS"); ok {
-		cmd.SetArgs(strings.Split(args, " "))
-	}
 
-	c.appDone = make(chan struct{})
 	go func() {
-		defer close(c.appDone)
```

Comment:
I think this might still be needed.  It is read from in `Stop()`.

Reply:
Good catch